cmd/dagger/logger/cloud2.go

```diff
@@ -0,0 +1,35 @@
+package logger
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"go.dagger.io/dagger/api"
+)
+
+type Cloud2 struct {
+	client *api.Client
+	url    string
+}
+
+func NewCloud2() *Cloud2 {
+	return &Cloud2{
+		client: api.New(),
+		url:    eventsURL(),
+	}
+}
+
+func (c *Cloud2) Write(p []byte) (int, error) {
```

Comment:
This is going to need a queue of some sort as it's synchronously blocking log writing (even on the terminal) and with a remote API server, the drag is going to be noticeable.

We can live with that for the first iteration. I think eventually we can use the pattern of #2515, rather than doing http requests here, we'd just call `telemetry.Push` which would handle the queueing and stuff

Reply:
Yup, definitely. I found some good stuff in https://github.com/dagger/dagger/blob/97c0aa634fe7ac78e0bf5bc5fd9aa5818d46f5e6/analytics/analytics.go